src/server/shared/DataStores/DBCDatabaseLoader.cpp

```diff
@@ -78,22 +78,20 @@ char* DBCDatabaseLoader::Load(uint32& records, char**& indexTable)
         uint32 indexValue = fields[_sqlIndexPos].GetUInt32();
         char* dataValue = indexTable[indexValue];
 
-        // If exist - override
-        newIndexes[newRecords] = indexValue;
-        dataValue = &dataTable[newRecords++ * _recordSize];
+        if (!dataValue)
+        {
+            newIndexes[newRecords] = indexValue;
+            dataValue = &dataTable[newRecords++ * _recordSize];
+        }
+        else
+            continue;
```

Comment:
Why **continue**? if you use "**continue**" here you won't load data from database unless it exists on dbc too. Isn't it?

I don't know if I'm right but, please, double check this part and try to explain me the logic